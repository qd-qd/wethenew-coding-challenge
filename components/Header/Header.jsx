import styles from './Header.module.css';

const Header = () => (
  <header className={styles.header} data-testid="header">
    <div className={styles['header-content']}>
      <img src="/logo.png" alt="logo" className={styles.logo} />
    </div>
  </header>
);

export default Header;
