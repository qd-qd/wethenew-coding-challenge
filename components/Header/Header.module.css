.header {
  display: flex;
  justify-content: center;
  width: 100vw;
  height: 52px;
  padding: 0 1.5rem;
  background-color: var(--theme-color-black);
}

.header-content {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  max-width: var(--theme-app-width);
}

.logo {
  height: 40px;
  width: 160px;
}

@media (min-width: 23.45em) {
  .header {
    height: 64px;
  }
}
