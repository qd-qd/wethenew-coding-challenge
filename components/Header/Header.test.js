import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Header from './Header';

describe('Header', () => {
  test('displays greeting', () => {
    render(<Header />);
    expect(screen.getByRole('img')).toHaveAttribute('alt');
    expect(screen.getByRole('img')).toHaveAttribute('src');
  });

  test("doesn't break snapshot", () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});
