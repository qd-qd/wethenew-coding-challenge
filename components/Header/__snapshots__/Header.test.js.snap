// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header doesn't break snapshot 1`] = `
<div>
  <header
    class="header"
    data-testid="header"
  >
    <div
      class="header-content"
    >
      <img
        alt="logo"
        class="logo"
        src="/logo.png"
      />
    </div>
  </header>
</div>
`;
